src/components/Auth/SignUp/SignUp.js

```diff
@@ -1,97 +1,95 @@
-import React, { useState } from 'react';
+import React from 'react';
 import {
   auth,
-  createUserProfileDocument,
+  signInWithGoogle,
 } from '../../../api/firebase/firebaseClient';
 import Input from '../../shared/Input/Input';
 import Button from '../../shared/Button/Button';
 import './SignUp.scss';
+import * as Yup from "yup";
+import {Field, Form, Formik} from "formik";
 
-const SignUp = () => {
-  const [userCredentials, setUserCredentials] = useState({
-    displayName: '',
-    email: '',
-    password: '',
-    confirmPassword: '',
-  });
-
-  const { displayName, email, password, confirmPassword } = userCredentials;
-
-  const handleSubmit = async (event) => {
-    event.preventDefault();
-    if (password !== confirmPassword) {
-      alert('hasła są niezgodne');
-      return;
-    }
-    try {
-      const { user } = await auth.createUserWithEmailAndPassword(
-        email,
-        password
-      );
-
-      setUserCredentials({
-        ...userCredentials,
-        displayName: '',
-        email: '',
-        password: '',
-        confirmPassword: '',
-      });
+const SignInSchema = Yup.object().shape({
+  email: Yup.string()
+      .typeError('Proszę podać poprawny adres email')
+      .email('Proszę podać poprawny adres email')
+      .required('Pole obowiązkowe')
+      .min(6, 'Minimum 6 znaków'),
+  password: Yup.string()
+      .min(6, 'Minimum 6 znaków')
+      .max(40, 'Maxymalnie 30 znaków')
+      .required('Pole obowiązkowe'),
+  confirmPassword: Yup.string()
+      .oneOf([Yup.ref('password'), null], 'Hasła muszą być identyczne')
+      .required('Pole obowiązkowe')
+});
```

Comment:
nie wiem czy nie dałbym minimum 8 znaków + 2 znaki specjalne? I moze jeszcze duza litera bedzie bardziej pro jezeli chodzi o walidacje hasla. Poza tym nazwa schema to SignIn a to jest komponent SignUp